lib/mongodb.js

```diff
@@ -2015,17 +2015,18 @@ function visitAllProperties(data, modelCtorOrDef, visitor) {
     const def = modelProps[p];
     if (def && def.type && isNestedModel(def.type)) {
```

Comment:
should we check with `value` is not null in this line instead of in line 2018 and 2026?

Reply:
Good point. Fixed.